supabase/functions/oauth/encrypt-config.ts

```diff
@@ -48,19 +50,23 @@ export async function encryptConfig(req: Record<string, any>) {
   });
 
   let responseData = JSON.stringify(await response.json());
-  console.log(responseData);
 
   // If we can find client_id or client_secret in plaintext in the response,
   // it's not secure to return this response!
   if (
-    responseData.includes(oauth2_client_id) ||
-    responseData.includes(oauth2_client_secret)
+    oauth2_client_id != null &&
+    oauth2_client_secret != null &&
+    (responseData.includes(oauth2_client_id) ||
+      responseData.includes(oauth2_client_secret))
   ) {
     return new Response(
       JSON.stringify({
-        error: "exposed_secret",
-        description: `client_id and client_secret were not encrypted as part of this request.
+        error: {
+          code: "exposed_secret",
+          message: `Request denied: "client id" and "client secret" could have been leaked.`,
+          description: `client_id and client_secret were not encrypted as part of this request.
 Make sure that they are marked with secret: true in the endpoint spec schema`,
+        },
```

Comment:
Nesting this feels dumb but it makes handling this a lot easier for the UI. I also switched the `error` key to `code` only because it is pretty close to what Supabase uses for Postgres errors and we can nicely display that in the UI without any coeding.